stages/org.osbuild.dracut

```diff
@@ -0,0 +1,147 @@
+  }
+  "add_modules": {
+    "description": "Additional dracut modules to include.",
+    "type": "array",
+    "items": {
+      "type": "string",
+      "description": "A dracut module, e.g. base, nfs, network ..."
+    }
+  },
+  "omit_modules": {
+    "description": "Dracut modules to not include.",
+    "type": "array",
+    "items": {
+      "type": "string",
+      "description": "A dracut module, e.g. base, nfs, network ..."
+    }
+  },
+  "drivers": {
+    "description": "Kernel modules to exclusively include.",
+    "type": "array",
+    "items": {
+      "type": "string",
+      "description": "A kernel module without the .ko extension"
+    }
+  },
+  "add_drivers": {
+    "description": "Add a specific kernel modules.",
+    "type": "array",
+    "items": {
+      "type": "string",
+      "description": "A kernel module without the .ko extension"
+    }
+  },
+  "force_drivers": {
+    "description": "Add driver and ensure that they are tried to be loaded.",
+    "type": "array",
+    "items": {
+      "type": "string",
+      "description": "A kernel module without the .ko extension"
+    }
+  },
+  "filesystems": {
+    "description": "Kernel filesystem modules to exclusively include.",
+    "type": "array",
+    "items": {
+      "type": "string",
+      "description": "A kernel module without the .ko extension"
+    }
+  },
+  "install": {
+    "description": "Add custom files to the initramfs.",
+    "type": "array",
+    "items": {
+      "type": "object",
+      "description": "What (keys) to include where (values)"
+    }
+  }
+}
+"""
+
+
+def main(tree, options):
+    modules = options.get("modules", [])  # dracut modules
+    add_modules = options.get("add_modules", [])
+    omit_modules = options.get("omit_modules", [])
+    drivers = options.get("drivers", [])  # kernel modules
+    add_drivers = options.get("add_drivers", [])
+    force_drivers = options.get("force_drivers", [])
+    filesystems = options.get("filesystems", [])
+    install = options.get("install", [])
+
+    # internal option for exploring purposes
+    host_only = options.get("host-only", False)
+
+    # initrds have already been created, force the recreation
+    options = ["--force", "-v", "--show-modules"]
+
+    options += ["--host-only" if host_only else "--no-host-only"]
+
+    if modules:
+        options += ["--modules", " ".join(modules)]
+
+    if add_modules:
+        options += ["--add", " ".join(add_modules)]
+
+    if omit_modules:
+        options += ["--omit", " ".join(omit_modules)]
+
+    if drivers:
+        options += ["--drivers", " ".join(drivers)]
+
+    if add_drivers:
+        options += ["--add-drivers", " ".join(add_drivers)]
+
+    if force_drivers:
+        options += ["--force-drivers", " ".join(force_drivers)]
+
+    if filesystems:
+        options += ["--filesystems", " ".join(filesystems)]
+
+    if install:
+        for l in install:
+            for k, v in l.items():
+                options += ["--install", k, v]
```

Comment:
I don't understand this option. The man page for dracut says that this is a simple list of file names..

Reply:
Ah yes, you are right, that should actually be `--include (-i)`; I guess `--install (-I)` is a special case of `--include`.